components/camel-direct/src/main/java/org/apache/camel/component/direct/DirectEndpoint.java

```diff
@@ -97,17 +94,11 @@ public void removeConsumer(DirectConsumer consumer) {
         }
     }
 
-    public void addProducer(DirectProducer producer) {
-        synchronized (consumers) {
-            producers.add(producer);
-        }
-    }
+    @Deprecated
```

Comment:
Its very likely fine to remove in he near future as nobody is extending direct component

Reply:
I have updated the pull request to remove the methods (not sure if I understood you correctly).